### ISSUE TITLE
[FEATURE REQ]Keep Open Telemetry attributes prefixed with db.comsosdb.* for dependency/span events

### ISSUE THREAD

**Issue by FabianMeiswinkel** [Azure/azure-sdk-for-java#34128]
**Is your feature request related to a problem? Please describe.**
Currently the applicationInsights-agent (3.4.10) is not adding attributes prefixed with db.cosmosdb.* to the dependency entity - see https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java. The .Net AI DependencyCollector is keeping them - which allows customers to see and query them. Is it possible to adjust the behavior in Java as well?

**Describe the solution you'd like**
Like in .Net - https://github.com/microsoft/ApplicationInsights-dotnet/blob/24e5b04e8f8a08de8efd1a973293e65914dd3df5/WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticsEventHandler.cs#L378 - keep the attributes prefixed with "db.cosmosdb.*" on the dependency entity in AI.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

**Comment by lmolkova** [comment]
@heyams @trask wdyt?

in .net we're adding all attributes (generic and db.cosmosdb) including `db.name`  that are transformed into AI schema to the custom dimensions. This allows cosmos folks to document telemetry in a vendor-agnostic way and support folks to know what specific things mean. Users can also query using `db.name`  without having a need to parse AppInsights target, name and other propeties. 

Would it be possible to follow the same approach in Java?

**Comment by heyams** [comment]
@lmolkova yes, we can do the same in Java to ensure feature parity. I can add it in the next exporter release.

**Comment by trask** [comment]
Discussed with @lmolkova, we are going to make the following proposal to all languages for the Azure Monitor OpenTelemetry exporters:

* Maintain list of specific attributes to exclude, e.g.
  * db.statement (because mapped 1-1)
  * net.transport (because noisy)
* All others attributes are mapped to custom dimensions

**Comment by FabianMeiswinkel** [comment]
@trask / @lmolkova - do you have any idea when you would be able to pick this up? Or would it make more sense for me or someone from the Cosmos DB SDK team to contribute to it?

**Comment by trask** [comment]
@FabianMeiswinkel sorry we lost track of this, I've just sent a PR to address it: https://github.com/Azure/azure-sdk-for-java/pull/42460
